org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/DiagnosticResolution.java

```diff
@@ -108,20 +111,12 @@ public String getLabel() {
 
 	public WorkspaceEdit apply() {
 		try {
-			Resource resource = options.getResource();
 			URI uri = resource.getURI();
-
-			CodeActionParams params = options.getCodeActionParams();
-			Document document = options.getDocument();
-			int offset = document.getOffSet(params.getRange().getStart());
-			EObjectAtOffsetHelper helper = new EObjectAtOffsetHelper();
-			ILanguageServerAccess access = options.getLanguageServerAccess();
 			ResourceSet resourceSet = access.newLiveScopeResourceSet(uri);
 			XtextResource tmpResource = (XtextResource) resourceSet.getResource(uri, true);
-			EObject obj = helper.resolveContainedElementAt(tmpResource, offset);
-
+			int offset = document.getOffSet(diagnostic.getRange().getStart());
+			EObject obj = new EObjectAtOffsetHelper().resolveContainedElementAt(tmpResource, offset);
```

Comment:
I know this code was present before, but `new EObjectAtOffsetHelper` might cause trouble since language specific customizing wouldn't be supported. Can we read the helper from the tmpResource's services?

Reply:
Sure, I will update the PS tomorrow, or if this one is already merged, create a new PR for only this.

Reply:
@szarnekow , I have done the change you suggested.